### PR TITLE
HHH-16815 Convenience methods for sorting and pagination

### PR DESCRIPTION
**Pull request by gavinking**
The final form of this API is:

```java
query.setOrder(Order.asc(Book_.title))
query.setOrder(Order.by(Book_.title, SortDirection.ASCENDING))
query.setOrder(List.of(Order.asc(Book_.title), Order.desc(Book_.isbn)))
query.setOrder(Order.asc(2))
query.setOrder(Order.by(2, SortDirection.ASCENDING))
query.setOrder(List.of(Order.asc(2), Order.desc(1)))
```

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java [hunk 1]
```diff
@@ -1085,7 +1091,7 @@ public Query<R> descending(int element) {
 	}
 
 	@Override
-	public Query<R> unordered() {
+	public Query<R> clearOrder() {
```
Comment (yrodiere) [inline hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java:1094]: Shouldn't this be `clearSort` if your other method is called `sort`?

Though TBH I find `clearOrder` + `orderBy(SingularAttribute, SortOrder)` would be closer to SQL/HQL and thus clearer/more consistent.

File: hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java [hunk 2]
```diff
@@ -493,6 +514,30 @@ default Stream<R> stream() {
 	@Incubating
 	SelectionQuery<R> descending(SingularAttribute<? super R, ?> attribute);
```
Comment (yrodiere) [inline hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java:515]: Shouldn't `ascending`/`descending` be removed in favor of `sort`? It feels like they no longer bring much...

File: hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java (hunk 1 above)
Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java:1094]: I agree. I prefer "order", which is what I started with. But then I would need to rename:

- `Sort` -> `Order`
- `SortOrder` -> ????

I need a new name for `SortOrder`.

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java:1094]: At one stage I tried calling it just `Direction` but that was a bit too generic. `OrderDirection` seems like it fits but for some reason I can't quite put my finger on it sounds weird to me. "Clumsy" as you say.

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java]: It's `SortDirection`.

File: hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java (hunk 2 above)
Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java]: @yrodiere You win. They are now gone.

File: hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSortSpecification.java [hunk 3]
```diff
@@ -52,7 +52,7 @@
 		return sortExpression;
 	}
 
-	public SortOrder getSortOrder() {
+	public SortDirection getSortOrder() {
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSortSpecification.java:55]: Hmm, this should probably be `getSortDirection` then

[the remark this answers is not in the document]
Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSortSpecification.java:55]: It's called `getDirection()` now, `githib-code-scanning` is slow.

[the remark this answers is not in the document]
Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSortSpecification.java:55]: Oh, no, wait, I lie, that's the method declared by `JpaOrder`. Hrm, OK, yes, should change that I suppose.

**Pull request merged**